app/utils/IncomeSourcesUtils.scala

```diff
@@ -40,6 +43,35 @@ trait IncomeSourcesUtils extends FeatureSwitching {
     }
   }
 
+  def newWithIncomeSourcesRemovedFromSession(redirect: Result, sessionService: SessionService, errorRedirect: Result)(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+    val incomeSourcesSessionKeys = Seq(
```

Comment:
we do have SessionKeys with all names of keys we have, so this would duplication;
if you need list of all key, you should define this in the SessionKeys object

Reply:
I have revamped the existing method withIncomeSourcesRemovedFromSession to handle the sessionService flow to remove from Mongo, Once all the IncomeSource sessions are moved to the sessionService we need to remove the duplicated method withIncomeSourcesRemovedFromSession.

Reply:
in incomeSourcesSessionKeys sequence, you should use variable defined in SessionKeys

Reply:
Changed to use SessionKeys.